### DISCUSSION TITLE
Wrong function called here?

### DISCUSSION THREAD

**Discussion by lubra123** [jmcdo29/testing-nestjs#1492] [Q&A]
https://github.com/jmcdo29/testing-nestjs/blob/f33ae43a95f85bd7517bfa69726bdd6b5482da4f/apps/mongo-sample/src/cat/cat.service.ts#L72

I could be wrong, but shouldn't you call the "delete" Function from the cat.controller.ts within this subproject? This can only work because this is a monorepo or am I missing something here?

**Comment by jmcdo29** [comment]
I'm not sure I understand what you mean. The `CatsController` calls the `CatsService`'s `deleteOne` method and the `CatsService` calls the `Model<Cat>`'s `remove` method

**Comment by TatyOko28** [comment]
The function call to this.catModel.remove({ id }) is correct if the intention is to delete a document directly at the database layer using the model.
However, if you want to align with the service-controller flow in NestJS, you should use the delete function from the cat.controller.ts, which likely wraps around the service logic.

✅ Key Points:

If direct database access is intended, calling this.catModel.remove() is fine.
If you follow a NestJS best practice, the controller should invoke the service, not the other way around.
